specification/digitaltwins/data-plane/Microsoft.DigitalTwins/stable/2020-10-31/digitaltwins.json

```diff
@@ -0,0 +1,1583 @@
+  },
+  "parameters": {
+    "eventRouteId": {
+      "name": "id",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The id for an event route. The id is unique within event routes and case sensitive.",
+      "x-ms-parameter-location": "method"
+    },
+    "includeModelDefinition": {
+      "name": "includeModelDefinition",
+      "description": "When true the model definition will be returned as part of the result.",
+      "in": "query",
+      "required": false,
+      "type": "boolean",
+      "default": false,
+      "x-ms-parameter-location": "method"
+    },
+    "digitalTwinId": {
+      "name": "id",
+      "in": "path",
+      "description": "The id of the digital twin. The id is unique within the service and case sensitive.",
+      "type": "string",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    },
+    "modelId": {
+      "name": "id",
+      "in": "path",
+      "description": "The id for the model. The id is globally unique and case sensitive.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "relationshipId": {
+      "name": "relationshipId",
+      "description": "The id of the relationship. The id is unique within the digital twin and case sensitive.",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "componentPath": {
+      "name": "componentPath",
+      "in": "path",
+      "description": "The name of the DTDL component.",
+      "type": "string",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    },
+    "api-version": {
+      "name": "api-version",
+      "in": "query",
+      "description": "The requested API version.",
+      "required": true,
+      "type": "string",
+      "enum": [
+        "2020-10-31"
+      ]
+    },
+    "max-items-per-page": {
+      "name": "max-items-per-page",
+      "in": "header",
+      "description": "The maximum number of items to retrieve per request. The server may choose to return less than the requested number.",
+      "required": false,
+      "type": "integer",
+      "x-ms-client-name": "MaxItemsPerPage",
+      "x-ms-parameter-location": "method",
+      "x-ms-parameter-grouping": {
+        "postfix": "Options"
+      }
+    },
+    "if-none-match-star": {
+      "name": "If-None-Match",
+      "in": "header",
+      "description": "Only perform the operation if the entity does not already exist.",
+      "required": false,
+      "type": "string",
+      "enum": [
+        "*"
+      ],
+      "x-ms-parameter-location": "method"
+    },
+    "if-match": {
+      "name": "If-Match",
+      "in": "header",
+      "description": "Only perform the operation if the entity's etag matches one of the etags provided or * is provided.",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "traceparent": {
```

Comment:
I'm not sure about these headers. There are standard haeders in this doc: https://www.w3.org/TR/trace-context/. You might want to reach out to a distributed tracing person (@lmolkova) to review this as its an industry standard.

Reply:
`traceparent` is a standard header according to https://www.w3.org/TR/trace-context/

Even if `tracestate` (another standard header for advanced cases)  is not applicable to DigitalTwins scenarios, I suggest to define it to fully comply with `tracestate`

Reply:
The service implements the W3C trace context standard. With approval from the Azure API review board, we added the trace context headers to the swagger. @lmolkova we hadn't added tracestate because the service doesn't support it, but I just learned that the service must at least propagate tracestate when its present. I'm following up on this now with the service team.

Reply:
The swagger has been updated to include `tracestate`.